src/wallet/api/wallet.cpp

```diff
@@ -513,22 +515,29 @@ string WalletImpl::keysFilename() const
     return m_wallet->get_keys_file();
 }
 
-bool WalletImpl::init(const std::string &daemon_address, uint64_t upper_transaction_size_limit)
+void WalletImpl::init(const std::string &daemon_address, uint64_t upper_transaction_size_limit)
 {
     clearStatus();
-    doInit(daemon_address, upper_transaction_size_limit);
-    bool result = this->refresh();
-    // enabling background refresh thread
-    startRefresh();
-    return result;
 
-}
+    m_wallet->init(daemon_address, upper_transaction_size_limit);
 
-void WalletImpl::initAsync(const string &daemon_address, uint64_t upper_transaction_size_limit)
-{
-    clearStatus();
-    doInit(daemon_address, upper_transaction_size_limit);
-    startRefresh();
+    // in case new wallet, this will force fast-refresh (pulling hashes instead of blocks)
+    // If daemon isn't synced a calculated block height will be used instead
+    if (isNewWallet() && daemonSynced()) {
+        LOG_PRINT_L2(__FUNCTION__ << ":New Wallet - fast refresh until " << daemonBlockChainHeight());
+        m_wallet->set_refresh_from_block_height(daemonBlockChainHeight());
+    }
+
+    if (m_rebuildWalletCache)
+      LOG_PRINT_L2(__FUNCTION__ << ": Rebuilding wallet cache, fast refresh until block " << m_wallet->get_refresh_from_block_height());
```

Comment:
This looks like a debug thing, so s/LOG_PRINT_L2/MDEBUG/.
At the top of a file, you define a default category:

#undef MONERO_DEFAULT_LOG_CATEGORY
#define MONERO_DEFAULT_LOG_CATEGORY "gui"

(or whatever is more granular if it makes sense; you can have gui.subcategory if you want to be able to set gui* and gui.subcategory separately).